mypy/plugins/functools.py

```diff
@@ -229,9 +229,11 @@ def partial_new_callback(ctx: mypy.plugin.FunctionContext) -> Type:
             partial_names.append(fn_type.arg_names[i])
         elif actuals:
             if any(actual_arg_kinds[j] == ArgKind.ARG_POS for j in actuals):
+                # Don't add params for arguments passed positionally
                 continue
+            # Add defaulted params for arguments passed via keyword
             kind = actual_arg_kinds[actuals[0]]
-            if kind == ArgKind.ARG_NAMED:
+            if kind == ArgKind.ARG_NAMED or kind == ArgKind.ARG_STAR2:
                 kind = ArgKind.ARG_NAMED_OPT
             partial_kinds.append(kind)
```

Comment:
What if `kind == ArgKind.STAR`? I mean in a situation like this
```python
def foo(x: int, y: int) -> None: ...
xs: list[int]
partial(foo, *xs)
```
IIUC this will create a callable with two star-args in signature, because star actual will map to both formals. I am not sure what is the best way to handle this (each way I can think of may create false positives), maybe we can at least handle the case where corresponding `arg_type` is fixed length tuple. But we definitely should not create invalid callables with multiple star-args.